shared/src/models/patient.ts

```diff
@@ -141,6 +124,14 @@ export class Patient {
     @Min(0)
     public readonly timeSpeed: number = 1;
 
+    /**
+     * Whether the {@link getVisibleStatus} of this patient has changed
+     * since the last time it was updated which personnel and materials treat him/her.
+     * Use this to prevent unnecessary recalculations for patients that didn't change -> performance optimization.
+     */
+    @IsBoolean()
+    public readonly visibleStatusChangedSinceTreatment: boolean = false;
```

Comment:
I couldn't come up with a better name. Therefore I would leave it like this.

Reply:
The name is long, true, but the name should not let somebody think something happens since treatment (treatment happens probably every second in the tick or not when patient has nobody assigned).

The name I wrote is more long, not coming up with a better name that is shorter, maybe:
`treatmentRecalculationSensible` (or usefull or ...), as before the recalculation wouldn't result in anything different.
But this name doesn't show that this is dependent on the visibleStatus

Reply:
The variable name should state what currently is and not what should be because you need the former information for a refactoring.

What about `visibleStatusChanged`?
There is a class comment that explains this variable more in detail, therefore this should be enough.

Reply:
> The variable name should state what currently is and not what should be because you need the former information for a refactoring.
> 
> What about `visibleStatusChanged`? There is a class comment that explains this variable more in detail, therefore this should be enough.

If the comment explains the rest (`visibleStatusChangedSinceLastTreatmentCalculation`), I would agree with this name, not the best but is not super long and better to finish than to discuss more time on this name.

Reply:
Renamed to `visibleStatusChanged`